### ISSUE TITLE
pip-compile throws error when building dependencies for rpy2

### ISSUE THREAD

**Issue by kurtwheeler** [jazzband/pip-tools#658]
I have had the package [rpy2](https://pypi.org/project/rpy2/) as a dependency for my project for a while. Recently (perhaps after upgrading pip?) trying to update my `requirements.txt` with `pip-compile requirements.in` has started resulting in an error. I have been able to reproduce the bug in a fresh virtualenv when the only dependency I'm trying to compile is `rpy2`.

##### Environment Versions

1. OS Type: Ubuntu 17.10
2. Python version:
```
$ python -V
Python 3.6.3
```
3. pip version:  
```
$ pip --version
pip 10.0.1 from /home/kurt/Development/test/test/lib/python3.6/site-packages/pip (python 3.6)
```
4. pip-tools version:
```
$ pip-compile --version
pip-compile, version 2.0.2
```

##### Steps to replicate

```
mkdir test
cd test
virtualenv test
source test/bin/activate
pip install pip-tools
echo "rpy2" > requirements.in
pip-compile requirements.in
```

##### Expected result

I expect `pip-compile` to output a requirements.txt file with all the dependencies needed for rpy2.

##### Actual result

It outputs:
```
Traceback (most recent call last):
  File "/home/kurt/Development/test/test/bin/pip-compile", line 11, in <module>
    sys.exit(cli())
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/piptools/scripts/compile.py", line 187, in cli
    results = resolver.resolve(max_rounds=max_rounds)
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/piptools/resolver.py", line 102, in resolve
    has_changed, best_matches = self._resolve_one_round()
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/piptools/resolver.py", line 199, in _resolve_one_round
    for dep in self._iter_dependencies(best_match):
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/piptools/resolver.py", line 285, in _iter_dependencies
    dependencies = self.repository.get_dependencies(ireq)
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/piptools/repositories/pypi.py", line 146, in get_dependencies
    self._dependencies_cache[ireq] = reqset._prepare_file(self.finder, ireq)
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/piptools/_vendored/pip/req/req_set.py", line 634, in _prepare_file
    abstract_dist.prep_for_dist()
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/piptools/_vendored/pip/req/req_set.py", line 129, in prep_for_dist
    self.req_to_install.run_egg_info()
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/piptools/_vendored/pip/req/req_install.py", line 439, in run_egg_info
    command_desc='python setup.py egg_info')
  File "/home/kurt/Development/test/test/lib/python3.6/site-packages/piptools/_vendored/pip/utils/__init__.py", line 707, in call_subprocess
    % (command_desc, proc.returncode, cwd))
pip.exceptions.InstallationError: Command "python setup.py egg_info" failed with error code 1 in /tmp/tmpro8i_9hlbuild/rpy2/
```

**Comment by Miserlou** [comment]
Confirmed with Python2.7 OSX as well.

**Comment by vphilippon** [comment]
@Miserlou In you case, can you check the version of `rpy2` used? The latest version `rpy2 2.9.3` doesn't support python 2.7:
```
$ pip install rpy2
<snip> 
    Complete output from command python setup.py egg_info:
    rpy2 is no longer supporting Python < 3. Consider using an older rpy2 release when using an older Python release.
```

@kurtwheeler There's a bug in `pip-tools` swallowing most of the setuptools failure output reported by pip, which makes debugging a bit harder.
Can you give a go to `pip install rpy2` and just to make sure that it actually works? I don't have R installed on my machine, which also cause this error.

If that `pip install` works fine, what might help us for now will be to hack pip-tools a bit: check https://github.com/jazzband/pip-tools/issues/654#issuecomment-382822832 , changing `if (logger.getEffectiveLevel ...` to `if(True)` to get more log details will help figuring out what's wrong in building the package.

**Comment by atugushev** [comment]
> There's a bug in `pip-tools` swallowing most of the setuptools failure output reported by pip, which makes debugging a bit harder.

FYI, it's fixed in `pip==19.1`. 

@kurtwheeler @Miserlou I know it's been some time, but could you please reproduce the bug with the latest pip?

**Comment by atugushev** [comment]
It's been some time so I will close this issue now, but please don't hesitate to reach out if you still have any issues. Thanks!
